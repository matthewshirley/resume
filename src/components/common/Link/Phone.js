import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const PhoneLink = ({ phoneNumber }) => {
  if (!phoneNumber) return (null);

  return (
    <Text as="a" href={`tel:${phoneNumber}`} m={0}>{phoneNumber}</Text>
  );
};

PhoneLink.propTypes = {
  phoneNumber: PropTypes.string,
};

PhoneLink.defaultProps = {
  phoneNumber: null,
};

export default PhoneLink;
