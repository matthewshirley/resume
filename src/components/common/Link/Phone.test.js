import React from 'react';
import renderer from 'react-test-renderer';

import Phone from './Phone';

describe('Phone Link', () => {
  it('renders correctly with an email address', () => {
    const tree = renderer
      .create(<Phone phoneNumber='1111111111' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does not render without an email', () => {
    const tree = renderer
      .create(<Phone phoneNumber={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
