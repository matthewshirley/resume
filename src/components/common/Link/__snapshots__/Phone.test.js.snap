// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phone Link does not render without an email 1`] = `null`;

exports[`Phone Link renders correctly with an email address 1`] = `
<a
  className="sc-AxjAm bnsefY"
  fontSize={1}
  href="tel:1111111111"
>
  1111111111
</a>
`;
